### PR TITLE
fix: (CXSPA-8044) - Fix "Show Review" button not scroll at first

### PR DESCRIPTION
**Pull request by petarmarkov9449**
Closes: [CXSPA-8044](https://jira.tools.sap/browse/CXSPA-8044)

"Show Review" button didn't scroll to "Review" tab at first.

### CODE REVIEW

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts [hunk 1]
```diff
@@ -72,12 +72,13 @@ export class ProductIntroComponent {
         const tabsComponent = this.getTabsComponent();
         const reviewsTab =
           tabsComponent && this.getTabByLabel(reviewsTabLabel, tabsComponent);
-
         if (reviewsTab) {
-          this.clickTabIfInactive(reviewsTab);
           setTimeout(() => {
-            reviewsTab.scrollIntoView({ behavior: 'smooth' });
-            reviewsTab.focus({ preventScroll: true });
+            this.clickTabIfInactive(reviewsTab);
```
Comment (Pio-Bar) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: Changes to the existing functionalities are considered breaking. You can see examples and how to deal with them here: https://wiki.one.int.sap/wiki/display/spar/Spartacus+Onboarding in the `Backwards Incompatibility` section. You can also look at other implementations of feature flags in the project e.g: `ScrollToTopComponent`

**Review by Pio-Bar — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts [hunk 2]
```diff
@@ -28,6 +34,11 @@ export class ProductIntroComponent {
   product$: Observable<Product | null> =
     this.currentProductService.getProduct();
 
+  //TODO: (CXSPA-8044) - remove feature flag next major release.
+  @Optional() protected featureConfigService = inject(FeatureConfigService, {
+    optional: true,
+  });
+
```
Comment (Pio-Bar) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]:
```suggestion
protected featureConfigService = inject(FeatureConfigService);
```

We've changed our approach slightly recently. The ticket number in the comment used to refer to a task reminding us to remove the feature flag eventually. Those have been deemed unnecessary and are no longer used.

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts [hunk 3]
```diff
@@ -64,6 +75,7 @@ export class ProductIntroComponent {
   /**
    * Scroll to views component on page and click "Reviews" tab
    */
+  //TODO: (CXSPA-8044) - remove feature flag next major release.
```
Comment (Pio-Bar) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]:
```suggestion
```

**Review by Pio-Bar — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts (hunk 3 above)
Comment (petarmarkov9449) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: I've removed it.

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts (hunk 2 above)
Comment (petarmarkov9449) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: I've updated.

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts (hunk 1 above)
Comment (petarmarkov9449) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: I've added a feature flag: a11yScrollToReviewByShowReview.

**Review by Pio-Bar — DISMISSED** [review]

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts [hunk 4]
```diff
@@ -28,6 +29,8 @@ export class ProductIntroComponent {
   product$: Observable<Product | null> =
     this.currentProductService.getProduct();
 
+  protected featureConfigService = inject(FeatureConfigService);
```
Comment (Zeyber) [inline projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: Lets change this to `private` as we don't want to expose it as a property to users of our libs. This makes it easier to remove later when we want to introduce breaking changes ;)

**Review by Zeyber — CHANGES_REQUESTED** [review]
Note to please be consistent with conventions we use on github as we have automations that rely on these. For this PR please:
- change branch name to use the format `feature/CXSPA-xxxx`. `fix/**` is not recognized by jobs.
- use "Closes: " instead of "Ticket: " in the description. We track this in changelogs.

File: projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts (hunk 4 above)
Comment (petarmarkov9449) [reply projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts]: I've changed it.

**Review by Zeyber — APPROVED** [review]

**Review by Zeyber — DISMISSED** [review]

**Review by developpeurweb — DISMISSED** [review]

**Review by Pio-Bar — APPROVED** [review]

**Pull request merged**